src/validation.cpp

```diff
@@ -4140,6 +4140,7 @@ void CChainState::LoadExternalBlockFile(FILE* fileIn, FlatFilePos* dbp)
                 std::deque<uint256> queue;
                 queue.push_back(hash);
                 while (!queue.empty()) {
+                    LOCK(cs_main);
```

Comment:
This is one of two places where ReadBlockFromDisk was not protected by cs_main.  PowCache is expecting to be protected externally, so we need this lock.